roles/wildfly_validation/tasks/pidfile_validate.yml

```diff
@@ -0,0 +1,21 @@
+---
+- name: "PID file stat"
+  ansible.builtin.stat:
+    path: "{{ wildfly_validation_systemd_pidfile }}"
+  register: pid_file_stat
+
+- name: "Verify PID file existence"
+  ansible.builtin.fail:
+    msg: "PID file {{ wildfly_systemd.selinux.pid_path }}/{{ wildfly_instance_name }}.pid does not exist."
+  ignore_errors: true
+  when: not pid_file_stat.stat.exists
+  tags:
+    - molecule-idempotence-notest
+
+- name: "Restart {{ wildfly_instance_name }} if PID file doesn't exit"
```

Comment:
I don't think a task in validation should change the state of the target. The idea of wildfly_validation is to be run, once the playbook has run and verify that everything is in order. If not, the validation role simply fails.
@guidograzioli any thoughts on that?

Reply:
That's good, then I'll not call `wildfly_validation` role from `wildfly_systemd` role.
Inside `wildfly_validation` it'll be only checking PID file exists or not, which is already [implemented](https://github.com/ansible-middleware/wildfly/blob/main/roles/wildfly_validation/tasks/systemd.yml#L19). not need to do anything extra.
and inside `wildfly_systemd` will check PID is there or not, if not then restart the service once. WDYT?

Reply:
> That's good, then I'll not call wildfly_validation role from wildfly_systemd role

No matter, roles should NOT call the validation's one. This one is explicitly optional and should not be triggered by any other one, even partially. If something needs to **perform** on the target, it should live in another role.

> Inside wildfly_validation it'll be only checking PID file exists or not, which is already [implemented](https://github.com/ansible-middleware/wildfly/blob/main/roles/wildfly_validation/tasks/systemd.yml#L19). not need to do anything extra.
 :+1: 
 Yep, just remove the service: tasks and your good on this front.
 
 > and inside wildfly_systemd will check PID is there or not, if not then restart the service once. WDYT?
 
 No, I think it's too hectic. IMHO, the best approach is to have the start of the service triggers an handler that check that the PID file was indeed created and has the appropriate metadata (ownership). Restarting the service because those are not correct will lead to some sort of infinite loop, because, most likely, the file will not be created on the second run.
 
Come to think of it,  I think we need to rework the handler to implement the following  steps:

1) wait for the PIDFile to be created (using the wait: I think there is a way to wait on file) to avoid having the handler fails because it was executed BEFORE the service had any chance to create the file... (this is need a rather short timeout to avoid making the playbook hang uselessly if the service did not start properly)

2) instead of using file (which potentially would change the target file), use stat: and assert: to verify that the PID file created has the appropriate metadata. If the pidfile does not respect the spec, the play simply fails.

In the _systemd role, you just add the notify to this handler, as you did previously, so it's triggered if service change from inactive to active.

@guidograzioli do you agree or did I forgot something?

Reply:
To be honest I'd skip completely on checking the pid in systemd role. We have to assume we deploy the correct configuration and that the pid will be created and with the correct permissions. 
Then, at a second time and out of the main playbook, we call the validation role to do all checks on it

Reply:
I tend to agree in fact... In the end, this a check, it just belongs in wildfly_validation.

Reply:
@rpelisse @guidograzioli changed it according to our last discussion.